utils/src/utils/credentialUtils.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { ApplicationSubscription } from '../../hostapi.v2';
+import { AzExtServiceClientCredentials, ISubscriptionContext } from '../../index';
+import { localize } from '../localize';
+
+/**
+ * Converts a VS Code authentication session to an Azure Track 1 & 2 compatible compatible credential.
+ */
+export function createCredential(getSession: (scopes?: string[]) => vscode.ProviderResult<vscode.AuthenticationSession>): AzExtServiceClientCredentials {
```

Comment:
We should note that this is more-or-less copied from Resource Groups. Is there an opportunity for code sharing or is this going to be too temporary to justify it?